### ISSUE TITLE
Feature Request: Parent-child process relationship

### ISSUE THREAD

**Issue by zbalkan** [wazuh/wazuh#17445]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.4.3 | Wazuh manager | Manager | N/A | N/A |

# Summary

As mentioned here (https://github.com/wazuh/wazuh/issues/16044), Sysmon rules create an alert when a suspicious process is initiated. However, it is not possible to omit the false positives with current setup. Because the event information does not include parent process. In order to verify if it is a suspicious event or a normal workflow, it is possible only if parent process information is queried within Indexer.

# Proposal

Just like the [process-forest](https://github.com/williballenthin/process-forest) tool does, it is possible to parse the Windows event logs and build a tree/forest of process creation events.

In order to provide detail on the need to have the parent-child process relationship in Wazuh, see the resources:

- https://twitter.com/SBousseaden/status/1365038669447524358/photo/3
- https://securityboulevard.com/2019/09/deobfuscating-ostap-trickbots-34000-line-javascript-downloader/
- https://tria.ge/210122-tx3pap8222/behavioral1
- https://twitter.com/Arkbird_SOLG/status/1352026888827531266
- https://twitter.com/Arkbird_SOLG/status/1276186169710174209

# Notes

It is hard to parse the event logs while storing them in memory in real-time, therefore it may not be handled with current engine as-is.

**Comment by mennotech** [comment]
I am also looking for parent process information, I believe that other security EDR solutions include this data in the logs. This helps for whitelisting alerts based on a trusted parent process.
